src/proc/Vigad.ts

```diff
@@ -1,5 +1,30 @@
-class Vigad {
-    public main() {
-        // ...
+import { StreamHandler } from './StreamHandler';
+
+export class Vigad {
+    private static instance: Vigad;
+    private streamHandler: StreamHandler;
+
+    /**
+     * Create a private constructor to prevent multiple instances
+     */
+    private constructor() {
+        this.streamHandler = StreamHandler.getInstance();
+    }
+
+    /**
+     * This function gets the singelton instance of the StreamHandler
+     *
+     * @returns the singelton instance of the StreamHandler
```

Comment:
```suggestion
     * @return Vigad
```
@param, @return, ... should always be specific (actual class names) as some IDEs work with these